src/main/java/com/origamisoftware/teach/agile/junit/SubscriptionPeriod.java

```diff
@@ -65,8 +78,11 @@ public int getTotalMonths() {
      * subscription period or false otherwise.
      */
     public boolean hasExpired(Date date) {
-        // todo implement method
-        throw new NotImplementedException();
+        /** todo implement method */
+       /**  throw new NotImplementedException(); */
+
+       boolean has_Expired =  endDate.after(date);
+        return has_Expired;
     }
```

Comment:
don't need to create a variable just to return it. 
sometimes that's a good idea if it makes the code more readable. 
most of the time is not necessary and it makes the computer do a little bit more work.